### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Mukola1983**
I deleted unused files.
Added new directory "cocktails" with new components

### CODE REVIEW

File: src/components/fetchTrain/Data/Data.jsx [hunk 1]
```diff
@@ -23,11 +23,14 @@ export default function Data(props) {
...
 
          {props.users.map((n) => {
-            return (
-               ((props.userId === 'users') && <p key={props.users.indexOf(n)}> <span>{n.name}</span>  {JSON.stringify(n.address)}</p>) ||
-               ((props.userId === 'posts') && <p key={props.users.indexOf(n)}><span>{getName(userNames, n.userId - 1)}</span> {n.title}</p>) ||
-               ((props.userId === 'todos') && <p key={props.users.indexOf(n)}>{n.title}</p>)
-            )
+
+            if (props.userId === 'users') {
+               return <p key={props.users.indexOf(n)}> <span>{n.name}</span>  {JSON.stringify(n.address)}</p>
```
Comment (serhiiminin) [inline src/components/fetchTrain/Data/Data.jsx:28]: it's better not to use the index of an array for React keys.   
When it's possible, it's better to use a unique key from `n` object
If it looks like this
```
const n = {
 id: 'id-23',
 title: 'Name',
}
```
Then use `n.id` as a key

File: src/components/cocktailsList/coctailsList/ingridients/Ingridients.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import PropTypes from 'prop-types'
+
+export default function Ingridients(props) {
+
+   const [active, setActive] = useState(false)
+
+   return (
+      <div className={styles.ingridients}>
+         <h3 >Ingridients</h3>
+         <button onClick={() => setActive(!active)}>{active ? "close" : "open"}</button>
```
Comment (serhiiminin) [inline src/components/cocktailsList/coctailsList/ingridients/Ingridients.jsx:13]: Do not introduce handlers right in the render section. This function will be recreated on each rerender.
Move it to a separate handler and wrap it with `useCallback`

**Review by serhiiminin — CHANGES_REQUESTED** [review]

File: src/components/cocktailsList/Cocktails.jsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+   // www.thecocktaildb.com/api/json/v1/1/search.php?i=vodka
+
+   return (
+
+      < div >
+         <h2>{title}</h2>
+         < CocktailInput startSearch={startSearch} search={search}
+            changeSearch={changeSearch} showTitle={showTitle}
+            changeQuery={changeQuery} />
+
```
Comment (Mukola1983) [inline src/components/cocktailsList/Cocktails.jsx:91]: Change on radiobutton

File: src/components/cocktailsList/Cocktails.jsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+            setErrorInp(true)
+            console.log("Something went wrong", error);
+         })
+   }, [showDrink, errorInp])
+
+   const changeSearch = useCallback((e) => {
+      setSearch(e.target.value)
+   }, [search]);
+
+
```
Comment (Mukola1983) [inline src/components/cocktailsList/Cocktails.jsx:41]: whether this is the correct use of the  "useCallback"

File: src/components/cocktailsList/Cocktails.jsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+      console.log(drinks)
+   }, [showDrink, search])
+
+   const showTitle = useCallback((letter, title) => {
+      setChangeQuery(letter)
+      setTitle(title)
+   }, [changeQuery])
+
+   {
+      if (errorInp) {
```
Comment (Mukola1983) [inline src/components/cocktailsList/Cocktails.jsx:62]: if the user enters an incorrect character, he will receive a warning

File: src/components/cocktailsList/Cocktails.jsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+            setErrorInp(true)
+            console.log("Something went wrong", error);
+         })
+   }, [showDrink, errorInp])
+
+   const changeSearch = e => {
+      setSearch(e.target.value)
+   }
+
+   const startSearch = e => {
```
[the remark this answers is not in the document]
Comment (Mukola1983) [reply src/components/cocktailsList/Cocktails.jsx]: changed

File: src/components/cocktailsList/Cocktails.jsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+         <h2>{title}</h2>
+         < CocktailInput startSearch={startSearch} search={search} changeSearch={changeSearch} />
+         <div>
+            <p>Search cocktail by name</p>
+            <button onClick={() => showTitle("s", "Search cocktail by name")}>search cbn</button>
+         </div>
+
+         <div>
+            <p>Search cocktails by first letter</p>
+            <button onClick={() => showTitle("f", "List all cocktails by first letter")}>search cbfl</button>
```
[the remark this answers is not in the document]
Comment (Mukola1983) [reply src/components/cocktailsList/Cocktails.jsx]: I remove this in "CocktailInput.jsx" and changed on radiobutton

File: src/components/cocktailsList/Cocktails.jsx [hunk 8]
```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useState } from 'react'
+import CocktailInput from './cocktailInput/CocktailInput';
+import CocktailsList from './coctailsList/CocktailsList';
+import Test from './Test';
+import { getApi } from './DataSource';
```
[the remark this answers is not in the document]
Comment (Mukola1983) [reply src/components/cocktailsList/Cocktails.jsx]: renamed

File: src/components/header/Header.jsx [hunk 9]
```diff
@@ -1,19 +1,17 @@
...
 
-import logo from './../../img/Avatars/logo192.png';
+import logo from "../../img/Avatars/logo192.png";
 
 import styles from './Header.module.css';
 
 
-const Header = () => {
-   return (
+const Header = () => (
```
[the remark this answers is not in the document]
Comment (Mukola1983) [reply src/components/header/Header.jsx]: changed in all problem files

File: src/components/footer/Footer.jsx [hunk 10]
```diff
@@ -4,14 +4,12 @@ import React from 'react';
 import styles from './Footer.module.css';
 
 
-const Footer = () => {
-   return (
+const Footer = () => (
       <header className={styles.footer}>
```
[the remark this answers is not in the document]
Comment (Mukola1983) [reply src/components/footer/Footer.jsx]: fixed

File: src/components/fetchTrain/Data/Data.jsx (hunk 1 above)
Comment (Mukola1983) [reply src/components/fetchTrain/Data/Data.jsx]: replaced on "n.id"

**Pull request merged**